### ISSUE TITLE
Full database sync fails in Container during database backup

### ISSUE THREAD

**Issue by timw** [orientechnologies/orientdb#9770]
### OS: Linux

When an OrientDb node running in a container attempts to perform a full sync, it fails during the database backup stage.

## Expected behavior  

During a full sync, the existing database is backed up and removed, allowing the full database sync to be performed in a new folder.

## Actual behavior  

While attempting to backup the current database, the backup fails with an exception in `Files.move`.

```
2022-03-09 13:52:42:303 INFO  Storage 'plocal:/orientdb/databases/indexity' is opened under OrientDB distribution : 3.1.13 - Veloce (build 825463fef9bb76863aac4653fa40f66f721786d9, branch UNKNOWN) [OLocalPaginatedStorage]
2022-03-09 13:52:43:299 INFO  [pmi-orientdb03] Requesting full sync for database 'indexity'... [OHazelcastPlugin]
2022-03-09 13:52:43:301 INFO  [pmi-orientdb03]->[[pmi-orientdb01]] Requesting deploy of database 'indexity' on local server... [OHazelcastPlugin]
2022-03-09 13:52:47:997 WARNI [pmi-orientdb03] Moving existent database 'indexity' in '/orientdb/databases/indexity' to '/orientdb/backup/databases/indexity' and get a fresh copy from a remote node... [OHazelcastPlugin]
2022-03-09 13:52:47:998 SEVER Atomic moves not supported during database backup, will try not atomic move [OHazelcastPlugin][pmi-orientdb03] Error on moving existent database 'indexity' located in '/orientdb/databases/indexity' to '/orientdb/backup/databases/indexity' (error=java.nio.file.DirectoryNotEmptyException: /orientdb/databases/indexity).
java.nio.file.DirectoryNotEmptyException: /orientdb/databases/indexity
	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:498)
	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
	at java.nio.file.Files.move(Files.java:1395)
	at com.orientechnologies.orient.server.distributed.impl.ODistributedAbstractPlugin.backupCurrentDatabase(ODistributedAbstractPlugin.java:1705)
	at com.orientechnologies.orient.server.distributed.impl.ODistributedAbstractPlugin.requestDatabaseFullSync(ODistributedAbstractPlugin.java:1497)
	at com.orientechnologies.orient.server.distributed.impl.ODistributedAbstractPlugin.requestFullDatabase(ODistributedAbstractPlugin.java:1276)
	at com.orientechnologies.orient.server.distributed.impl.ODistributedAbstractPlugin$4.call(ODistributedAbstractPlugin.java:1107)
	at com.orientechnologies.orient.server.distributed.impl.ODistributedAbstractPlugin$4.call(ODistributedAbstractPlugin.java:1013)
	at com.orientechnologies.orient.server.distributed.impl.ODistributedAbstractPlugin.executeInDistributedDatabaseLock(ODistributedAbstractPlugin.java:1963)
	at com.orientechnologies.orient.server.distributed.impl.ODistributedAbstractPlugin.installDatabase(ODistributedAbstractPlugin.java:1009)
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.installNewDatabasesFromCluster(OHazelcastPlugin.java:1664)
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.startup(OHazelcastPlugin.java:315)
	at com.orientechnologies.orient.server.OServer.registerPlugins(OServer.java:1258)
	at com.orientechnologies.orient.server.OServer.activate(OServer.java:500)
	at com.orientechnologies.orient.server.OServerMain$1.run(OServerMain.java:49)

```

## Analysis

The OrientDB Docker containers define 3 distinct volumes:
 - databases
 - backup
 - config

During a full sync, the current database is first moved out of the way using `ODistributedAbstractPlugin.backupCurrentDatabase`, which in turn invokes `Files.move`.
`Files.move` is documented to specifically only work where:
 - The folder can be moved by renaming
 - or, the source is an empty directory or a symlink

In a container, the `databases` and `backup` locations are on distinct volumes, so this operation always fails.

This could be seen as just an issue with the Docker container (i.e. the volume strategy in the container needs to change to support this), or it could be improved to copy and then remove the database folder (although I'm not sure if this is a strategy I'd want to deploy for any real-world data scale), so raising this as an issue for discussion here first.

Since you can't `undeclare` volumes when extending Docker containers, it's messy to override the volume setup in our own container, so an upstream fix would be better.

**Comment by tglman** [comment]
Hi @timw,

Thanks for the detailed analysis, do you have already a possible replacement solution for the problem ? if you do feel free  to open a PR, otherwise we will check it on our side.

Regards

**Comment by tglman** [comment]
Hi,

This should be fixed in all the 3.0.x, 3.1.x and 3.2.x and will be released in the next hotfix of each series, will leave it open till released

**Comment by tglman** [comment]
released for both 3.1.x and 3.2.x series, closing
